firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java

```diff
@@ -525,6 +525,7 @@ private ListenerRegistration addSnapshotListenerInternal(
   }
 
   @Override
+  @SuppressWarnings("EqualsGetClass") // This class is effectively final.
```

Comment:
Question: why not use `instanceof` here and in `Query#equals`? The article you linked states that `instanceof` is never worse than `getClass`:

> What about final or effectively-final classes?
> 
> In this case there is no disadvantage to the `getClass` trick - but there’s no great advantage to it either. Most unsafe idioms have circumstances in which they are safe, but this doesn’t change the fact that they are generally unsafe and not worth propagating and legitimizing.